src/c/c_dataset.cpp

```diff
@@ -30,163 +30,196 @@
 
 using namespace SmartRedis;
 
+// Create a new DataSet.
+// The user is responsible for deallocating the DataSet via DeallocateeDataSet()
 extern "C"
-void* CDataSet(const char* name, const size_t name_length)
+void *CDataSet(const char *name, const size_t name_length)
 {
-  /* Return a pointer to a new DataSet.
-  The user is responsible for deleting the client.
-  */
-  std::string name_str = std::string(name, name_length);
-  DataSet* dataset = new DataSet(name_str);
-  return (void*)dataset;
+  // Sanity check parameters
+  if (NULL == name)
+    return NULL;
+
+  std::string name_str(name, name_length);
+
+  DataSet *dataset = NULL;
+  try {
+    dataset = new DataSet(name_str);
+  } catch (...) {
+    dataset = NULL;
+  }
+  
+  return reinterpret_cast<void *>(dataset);
 }
 
+// Deallocate a DataSet
+extern "C"
+void DeallocateeDataSet(void *dataset)
+{
+  // Sanity check parameters
+  if (NULL == dataset)
```

Comment:
Maybe this can be condense to remove mid-function ``return`` statement with:

```
 if (NULL != dataset)
           delete reinterpret_cast<DataSet *>(dataset)
```

Reply:
When every routine starts with a sanity check of this form, it becomes a sort of automatic checklist item. Consistency across the codebase helps. I'd rather not change this, but we can discuss